### PR TITLE
Xml Warning Cleanup

### PR DESCRIPTION
**Pull request by cemremengu**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [x] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [x] I have provided test coverage for my change (where applicable)

### Description
This pull request eliminates the warnings caused by missing XML comments/docs for the files. All files were documented with the guidance of GhostDoc. Only 3 files that are maintained externally are omitted. The XML related warnings in these files are supressed by:

`#pragma warning disable CS1591, CS1574, CS1711, CS1712`

These files are as follows:

- TinyIoC.cs
- HttpUtility.cs
- SimpleJson.cs

### CODE REVIEW

File: src/Nancy/Responses/EmbeddedFileResponse.cs [hunk 1]
```diff
@@ -75,7 +84,7 @@ private static string ByteArrayToString(byte[] data)
             var output = new StringBuilder(data.Length);
             for (int i = 0; i < data.Length; i++)
             {
-                output.Append(data[i].ToString("X2"));
+                output.Append(i.ToString("X2"));
```
Comment (xt0rted) [inline src/Nancy/Responses/EmbeddedFileResponse.cs]: This change is a bug and is why some of the unit tests are failing now

File: src/Nancy/Diagnostics/DiagnosticsSession.cs [hunk 2]
```diff
@@ -4,12 +4,14 @@ namespace Nancy.Diagnostics
...
     using System.Text;
 
-#if !NETSTANDARD1_5
-    [Serializable]
-#endif
     /// <summary>
     /// Stores the http session information for diagnostics.
     /// </summary>
+#if !NETSTANDARD1_5
+
```
Comment (xt0rted) [inline src/Nancy/Diagnostics/DiagnosticsSession.cs]: There's extra whitespace now

**Comment by xt0rted** [comment]
Aside from that one bug that was introduced nothing stood out as being an issue.

There were a few spots where if statements were replaced with the ternary operator which, for me personally, makes the code a bit harder to read despite being functionally identical.

**Comment by cemremengu** [comment]
@xt0rted Thanks for the quick review. I have fixed the bug and removed the new line. Local tests and travis-ci are green now but appveyor and TC is still giving me a red. I checked the appveyor logs and cant see what exactly is failing. Any ideas ?

**Comment by xt0rted** [comment]
The same issue came up in #2492 but I'm not sure if anyone has looked into it yet.

**Comment by jchannon** [comment]
@cemremengu this is very impressive! Is this a Resharper clean type function?  One thing I think we need to look out for is any code changes, @xt0rted mentioned ternary logic was added.  Because this changes so many files its going to be hard to review all of them so if we are confident its just XML added and not code changes it would be easier

**Comment by cemremengu** [comment]
@jchannon All changes were run manually under my supervision so I double checked `GhostDoc` and `Resharper` outputs. Ternary changes weren't too many though; 3-4 as far as I remember. (I like ternary btw :smile:) I can revert any code changes as you want so just let me know.

**Comment by jchannon** [comment]
I think the ternary changes should be reverted for now. That can be done in
another PR. That way we can be sure this is just XML changes

**Pull request closed without merging**